HelpSource/Tutorials/Getting-Started/04-Functions-and-Other-Functionality.schelp

```diff
@@ -39,49 +53,45 @@ f.value;
+(
+f = {
+	"Evaluating...".postln;
+	2 + 3;
+};
 f.value;
+)
 ::
 
-Often methods simply return the object itself. This is the case with most objects and the message 'value'. The example below demonstrates this. (Everything to the right of the code:://:: is a 'comment', which means that SC just ignores it. Comments are a good idea to make your code clearer.)
-
-code::
-f = 3;			// Here I make f equal to a number
-f.value;		// Post window says: 3, i.e it returns itself
-f.value;		// Still says 3
-
-f = { 3.0.rand; };	// Here it's a Function.
-f.value;		// 3.0.rand means return a random value from 0 to 3.0 exclusive.
-f.value;		// something different
-f.value;		// something different
-::
-
-This means that by using the 'value' method Functions and other objects can be interchangeable in your code. This is an example of emphasis::polymorphism::, which is one of the powerful features of what's called Object Oriented Programming. Polymorphism just means that different objects are interchangeable (at least providing they return something sensible for what you're doing) if they respond to the same message. Object Oriented Programming (or OOP, as it's called for short) just means programming with objects. Simple, yes? Here's another short example showing this in action:
+Often methods simply return the object itself. This is the case with most objects and the message "value". The example below demonstrates this. (Everything to the right of the code:://:: is a "comment", which means that SC just ignores it. Comments are often used to leave notes on complicated or confusing pieces of code for future readers)
 
 code::
-f = { arg a; a.value + 3 };	// call 'value' on the arg; polymorphism awaits!
-f.value(3);			// 3.value = 3, so this returns 3 + 3 = 6
-g = { 3.0.rand; };
-f.value(g);			// here the arg is a Function. Cool, huh?
-f.value(g);			// try it again, different result
+f = 3;			// Create a variable and assign it a number as a value
+f.value;		// Post window says: 3, i.e the number returns itself
+f.value;		// f hasn't changed, so the post window says 3 again
+
+f = { 3.0.rand; };	// Assign the variable a Function as a value
+f.value;		// 3.0.rand returns a random value from 0.0 to 3.0 (exclusive).
+f.value;		// Another random value
+f.value;		// Yet another random value
 ::
 
-Start to see how this could be useful?
+By using the "value" method Functions and other objects can be interchangeable in your code. This is an example of emphasis::polymorphism::, which is one of the powerful features of what's called Object Oriented Programming (OOP). Polymorphism means that different objects are interchangeable if they respond to the same message. More on this later.
 
 Functions can also have what are called arguments. These are values which are passed into the Function when it is evaluated. The example below demonstrates how this works. See if you can guess what the result will be before executing it.
 
 code::
 (
-f = { arg a, b;
+f = {
+	arg a, b;
```

Comment:
perhaps write on the format: `f = { |a, b| … ) ` ?

Reply:
That notation is introduced down on line ~146: "There is an alternate way to specify args ..."

Reply:
the pipe notation is what our own style guide suggestions, i think we should present it first. maybe outside the scope of this PR

Reply:
I haven't changed the order of `arg` vs pipes as I feel that should be a more thorough review of our codebase... If you feel strongly it should be included in this PR, I can make that change.

Reply:
ok, that is fine with me.